openpype/hosts/maya/plugins/publish/validate_rendersettings.py

```diff
@@ -275,16 +275,16 @@ def get_invalid(cls, instance):
         # go through definitions and test if such node.attribute exists.
         # if so, compare its value from the one required.
         for attribute, data in cls.get_nodes(instance, renderer).items():
-            # Validate the settings has values.
-            if not data["values"]:
-                cls.log.error(
-                    "Settings for {}.{} is missing values.".format(
-                        node, attribute
+            for node in data["nodes"]:
+                # Validate the settings has values.
+                if not data["values"]:
+                    cls.log.error(
+                        "Settings for {}.{} is missing values.".format(
+                            node, attribute
+                        )
```

Comment:
Seems a bit much that we're reporting this _per node_ just because the setting value is empty?

It might make much more sense to just IGNORE the value already in `get_nodes` method if it doesn't have any values and then log a warning/error once that it's skipping that particular attribute?

That'd also help to even avoid querying for the nodes too.

Reply:
Yup good point.